modules/search-pipeline-common/src/yamlRestTest/resources/rest-api-spec/test/search_pipeline/40_rename_response.yml

```diff
@@ -0,0 +1,59 @@
+---
+teardown:
+  - do:
+      search_pipeline.delete:
+        id: "my_pipeline"
+        ignore: 404
+
+---
+"Test filter_query processor":
+  - do:
+      search_pipeline.put:
+        id: "my_pipeline"
+        body: >
+          {
+            "description": "test pipeline",
+            "response_processors":[
+              {
+                "rename":
+                  {
+                    "field": "a",
+                    "target_field": "b"
+                  }
+              }
+            ]
+          }
+  - match: { acknowledged: true }
+
+  - do:
+      index:
+        index: test
+        id: 1
+        body: {
+          "a": "foo"
+        }
+
+  - do:
+      indices.refresh:
+        index: test
+
+  - do:
+      search:
+        body: { }
+  - match: { hits.total.value: 1 }
+
+  - do:
+      search:
+        index: test
+        search_pipeline: "my_pipeline"
+        body: { }
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0._source: {"b": "foo"} }
+
+  # Should also work with no search body specified
+  - do:
+      search:
+        index: test
+        search_pipeline: "my_pipeline"
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0._source: {"b": "foo"} }
```

Comment:
Please also include a tests for the following:
1. Search request body has `"_source":false, `"stored_fields":["a"]`
2. Search request body has `"_source":false, `"docvalue_fields":["a"]`
3. Tests covering missing fields (with and without `"ignore_missing":true`).

Reply:
Addressed in [04614f0](https://github.com/opensearch-project/OpenSearch/pull/7377/commits/04614f0120c55df1fbd5c33d86b5c947ee8b4ed2)